sdk/notificationhubs/notification-hubs/src/api/beginSubmitNotificationHubJob.ts

```diff
@@ -117,30 +120,45 @@ export async function beginSubmitNotificationHubJob(
       return () => progressCallbacks.delete(s);
     },
 
-    isDone(): boolean {
+    get isDone(): boolean {
       return ["succeeded", "failed", "canceled"].includes(state.status);
     },
 
-    stopPolling(): void {
-      abortController.abort();
-      cancelJob?.();
+    get operationState(): OperationState<NotificationHubJob> | undefined {
+      return state;
     },
 
-    isStopped(): boolean {
-      return resultPromise === undefined;
+    get result(): NotificationHubJob | undefined {
+      return state.result;
     },
 
-    getOperationState(): OperationState<NotificationHubJob> {
-      return state;
+    async serialize(): Promise<string> {
+      return JSON.stringify({ state });
```

Comment:
I was wondering if customers only created this poller and didn't call any poll or pollUntilDone to update the state.

Then they would get the string like `{
    status: "notStarted",
  }`?

Reply:
I think so, but I don't think that's incorrect. As I understand, the state will reflect the state at a given instant, so when creating the Poller the status is "notStarted" if we don't poll, the status is true.

I think the decision to poll when the poller is created is something each library may want to make. Specially if they implement their pollers like here. In codegen we decided to poll immediatly after creation, however there are 2 things to consider in Notification Hubs. I don't think aligning to the core implementation is a must, especially since the current implementation has been released.


1. Would changing the current implementation to poll automatically after creation add value to existing customers?
2. How disruptive this change would be for existing customers. Even if we are taking a breaking change, we need to make sure we don't cause unecessary friction.